app/domain/authentication/authn_jwt/parse_claim_path.rb

```diff
@@ -2,22 +2,15 @@ module Authentication
   module AuthnJwt
 
     # This class parses complex claim path string
-    # like claim1/claim2[3]/claim3[4][67]/claim6
+    # like claim1/claim2/claim3/claim6
     # to array where claim names are strings and indexes are ints
     class ParseClaimPath
-      DEFAULT_PATH_SEPARATOR = "/".freeze
-      SINGLE_CLAIM_NAME_REGEX = /[a-zA-Z|$|_][a-zA-Z|$|_|0-9|.]*(\[\d+\])*/.freeze
-      NESTED_CLAIM_NAME_REGEX = %r{^#{SINGLE_CLAIM_NAME_REGEX.source}(#{DEFAULT_PATH_SEPARATOR}#{SINGLE_CLAIM_NAME_REGEX.source})*$}.freeze
-
-      def call(claim:, parts_separator: DEFAULT_PATH_SEPARATOR)
-        raise Errors::Authentication::AuthnJwt::InvalidClaimPath.new(claim, NESTED_CLAIM_NAME_REGEX) if
-          claim.nil? || !claim.match?(NESTED_CLAIM_NAME_REGEX)
+      def call(claim:, parts_separator: PATH_DELIMITER)
```

Comment:
Please add debug logs

Reply:
Do you think that each pure function should has its own log?

Reply:
i think we should have a log that we are parsing a value to another format